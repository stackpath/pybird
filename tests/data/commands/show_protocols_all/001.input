
0001 BIRD 2.0.4 ready.
2002-Name       Proto      Table      State  Since         Info
1002-bfd1       BFD        ---        up     2019-06-03
1006-
1002-pipe1      Pipe       ---        up     2019-06-03    foo_net <=> master4
1006-  Channel main
     Table:          foo_net
     Peer table:     master4
     Import state:   up
     Export state:   up
     Import filter:  REJECT
     Export filter:  (unnamed)
   Routes:         0 imported, 4 exported
   Route change stats:     received   rejected   filtered    ignored   accepted
     Import updates:              5          5          0          0          0
     Import withdraws:            0          0        ---          0          0
     Export updates:             10          0          5          1          4
     Export withdraws:            0          0        ---          0          0

1002-link_aaa  BGP        ---        up     2019-06-03    Established
1006-  Description:    Hub peer session - link_aaa
   BGP state:          Established
     Neighbor address: 192.168.7.0
     Neighbor AS:      4220000000
     Neighbor ID:      10.0.10.1
     Local capabilities
       Multiprotocol
         AF announced: ipv4
       Route refresh
       Graceful restart
         Restart time: 120
         AF supported: ipv4
         AF preserved:
       4-octet AS numbers
       Enhanced refresh
       Long-lived graceful restart
     Neighbor capabilities
       Multiprotocol
         AF announced: ipv4
       Route refresh
       Graceful restart
         Restart time: 120
         AF supported: ipv4
         AF preserved:
       4-octet AS numbers
       Enhanced refresh
       Long-lived graceful restart
     Session:          external AS4
     Source address:   192.168.1.1
     Hold timer:       115.850/180
     Keepalive timer:  10.618/60
   Channel ipv4
     State:          UP
     Table:          foo_net
     Preference:     100
     Input filter:   (unnamed)
     Output filter:  (unnamed)
     Import limit:   25000
       Action:       disable
     Routes:         3 imported, 0 filtered, 2 exported, 3 preferred
     Route change stats:     received   rejected   filtered    ignored   accepted
       Import updates:              3          0          0          0          3
       Import withdraws:            0          0        ---          0          0
       Export updates:              7          3          2        ---          2
       Export withdraws:            0        ---        ---        ---          0
     BGP Next hop:   192.168.1.1
     IGP IPv4 table: foo_net

0000