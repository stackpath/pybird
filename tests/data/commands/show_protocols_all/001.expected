[
    {"export_updates_received": 7, 
    "protocol": "BGP", 
    "export_updates_rejected": 3, 
    "import_updates_ignored": 0, 
    "export_updates_accepted": 2, 
    "import_updates_accepted": 3, 
    "routes_imported": 3, 
    "routes_preferred": 3, 
    "routes_filtered": 0, 
    "export_withdraws_accepted": 0, 
    "import_withdraws_accepted": 0, 
    "export_withdraws_received": 0, 
    "state": "Established", 
    "import_withdraws_received": 0, 
    "router_id": "10.0.10.1", 
    "description": "Hub peer session - link_aaa", 
    "import_withdraws_rejected": 0, 
    "import_updates_received": 3, 
    "export_updates_filtered": 2, 
    "address": "192.168.7.0", 
    "import_withdraws_ignored": 0, 
    "asn": "4220000000", 
    "import_updates_filtered": 0, 
    "name": "link_aaa", 
    "up": true, 
    "routes_exported": 2, 
    "last_change": "2019-06-03T00:00:00", 
    "import_updates_rejected": 0,
    "ipv4_state": "UP"
    }
]